client/signup/main.jsx

```diff
@@ -149,7 +151,14 @@ const Signup = React.createClass( {
 			defer( function() {
 				page( destination );
 			}.bind( this ) );
-		} else {
+		}
+
+		if ( ! user.get() && config.isEnabled( 'oauth' ) ) {
```

Comment:
Although, I would suggest setting the computed values of the `user.get` and `config.isEnabled` functions in the `if` to variables so we're not repeatedly calling them.